great_expectations/datasource/fluent/interfaces.py

```diff
@@ -514,21 +510,6 @@ def parse_order_by_sorters(
                     order_by_sorters.append(sorter)
         return order_by_sorters
 
-    @staticmethod
-    def parse_batching_regex_string(
-        batching_regex: Optional[Union[re.Pattern, str]] = None
-    ) -> re.Pattern:
-        pattern: re.Pattern
-        if not batching_regex:
-            pattern = MATCH_ALL_PATTERN
-        elif isinstance(batching_regex, str):
-            pattern = re.compile(batching_regex)
-        elif isinstance(batching_regex, re.Pattern):
-            pattern = batching_regex
-        else:
-            raise ValueError('"batching_regex" must be either re.Pattern, str, or None')
-        return pattern
```

Comment:
This is now all handled by the `batching_regex` default pydantic type coercion and the default field value of `".*"`